docs/details/main-components/timer.rst

```diff
@@ -46,74 +64,110 @@ For example:
 :cpp:func:`lv_timer_handler`.
 
-:cpp:expr:`lv_timer_reset(timer)` resets the period of a timer. It will be
-called again after the defined period of milliseconds has elapsed.
+:cpp:expr:`lv_timer_reset(timer)` resets the period of a Timer. It will be
+called again after its currently-set period (in milliseconds) has elapsed.
+
 
-Set parameters
-**************
 
-You can modify some timer parameters later:
+Setting Parameters
+******************
+
+You can modify these Timer parameters at any time during its life:
 
 - :cpp:expr:`lv_timer_set_cb(timer, new_cb)`
-- :cpp:expr:`lv_timer_set_period(timer, new_period)`
+- :cpp:expr:`lv_timer_set_period(timer, new_period_ms)`
+- :cpp:expr:`lv_timer_set_user_data(timer, user_data)`
+
+
 
-Repeat count
+Repeat Count
 ************
 
-You can make a timer repeat only a given number of times with
-:cpp:expr:`lv_timer_set_repeat_count(timer, count)`. The timer will
-automatically be deleted after it's called the defined number of times.
-Set the count to ``-1`` to repeat indefinitely.
+When a Timer is created, its repeat-count is set to ``-1`` to cause it to repeat
+indefinitely.  You can make a Timer repeat only a given number of times with
+:cpp:expr:`lv_timer_set_repeat_count(timer, count)`.  By default, once the Timer has
+run ``count`` times, it will be automatically deleted.
+
+You can use :cpp:expr:`lv_timer_set_auto_delete(timer, false)` if you want the timer
+to instead be paused after it has run ``count`` times.  This can be handy if you
+reuse that timer repeatedly and want to avoid the CPU and :cpp:func:`lv_malloc`
+overhead of repeatedly creating and deleting a timer.  If you use this option, you
+will need to set its repeat count (to either ``-1`` or a positive repeat count, since
+it will have decremented to ``0``) and :ref:`resume <timer_pause_and_resume>` it to
+make it active again.
+
 
-Enable and Disable
-******************
 
-You can enable or disable a timer with :cpp:expr:`lv_timer_enable(en)`.
+.. _timer_pause_and_resume:
 
 Pause and Resume
 ****************
 
-:cpp:expr:`lv_timer_pause(timer)` pauses the specified timer.
+:cpp:expr:`lv_timer_pause(timer)` pauses the specified Timer.
 
-:cpp:expr:`lv_timer_resume(timer)` resumes the specified timer.
+:cpp:expr:`lv_timer_resume(timer)` resumes the specified Timer.
 
-Measure idle time
-*****************
+
+
+Measuring Idle Time
+*******************
 
 You can get the idle percentage time of :cpp:func:`lv_timer_handler` with
 :cpp:func:`lv_timer_get_idle`. Note that, it doesn't measure the idle time of
-the overall system, only :cpp:func:`lv_timer_handler`. It can be misleading if
-you use an operating system and call :cpp:func:`lv_timer_handler` in a timer, as
-it won't actually measure the time the OS spends in an idle thread.
+the overall system, only :cpp:func:`lv_timer_handler`.  This can be misleading if
+you are using an operating system, as it does not actually measure the time the OS
+spends in an idle thread.
+
```

Comment:
IMO we can mention [`LV_SYSMON_GET_IDLE`](https://github.com/lvgl/lvgl/blob/master/lv_conf_template.h#L952-L953) here. It allow to use an OS-aware `get_idle` function.

Reply:
> IMO we can mention [`LV_SYSMON_GET_IDLE`](https://github.com/lvgl/lvgl/blob/master/lv_conf_template.h#L952-L953) here. It allow to use an OS-aware `get_idle` function.

I'm unaware of how it works.  Would you mind writing what you know about it and I can incorporate it into `timer.rst`?

Reply:
Sure! 

[Here](https://github.com/lvgl/lvgl/blob/master/src/others/sysmon/lv_sysmon.c#L249) we use `LV_SYSMON_GET_IDLE` to get the idle percentage. By default `LV_SYSMON_GET_IDLE` is set the `lv_timer_get_idle`, but if there is an operating system and DMA/GPU is used during rendering it can happen that rendering took 20 ms, but from this 10 ms was spend by waiting for DMA/GPU mutex. So although it was in `lv_timer_handler` it was free and the OS could run other tasks. 

For FreeRTOS I've already added [a special idle get function](https://github.com/lvgl/lvgl/blob/master/src/osal/lv_freertos.c#L393-L422) which returns the idle based on the time spent in the idle task. It can be used in `LV_SYSMON_GET_IDLE` to get the the OS idle instead.

Reply:
> Sure!

Done and submitted!

I observed that instead of writing what you would tell the reader, you made me go and look at the C code.  \*\*smiling\*\*  I can appreciate the value in that:  I'm learning more and more about the components that are available.  And so many of them are not documented at all (even in the source code).  :-(  I'm hoping we can remedy that at some point.

Reply:
>  I'm hoping we can remedy that at some point.

Me too :smile: